tests/integration/tests/test_sanity_checks.py

```diff
@@ -304,18 +304,29 @@ def wait_components_list(actual: AgentComponentMapping, expected: AgentExpectedM
     if not elasticsearch.kibana_url:
         return [""]
 
-    actual.load_map()
+    try_loading_map(actual)
```

Comment:
@gurevichdmitry This is the fix. When we call `actual.load_map()` too soon we will get a response, but it will be missing `components` key. I've wrapped the method to handle errors like this and now it successfully awaits `actual` to be populated. Not sure why this came up just now, but it's all in timing.